tests/specs/session-replay/ingest.e2e.js

```diff
@@ -0,0 +1,50 @@
+import { config, getSR } from './helpers'
+import { testBlobRequest } from '../../../tools/testing-server/utils/expect-tests'
+
+export default (function () {
+  describe('Session Replay Ingest Behavior', () => {
+    afterEach(async () => {
+      await browser.destroyAgentSession(browser.testHandle)
+    })
+
+    it('Should empty event buffer when sending', async () => {
+      await browser.url(await browser.testHandle.assetURL('rrweb-instrumented.html', config()))
+        .then(() => browser.waitForAgentLoad())
+
+      expect((await getSR()).events.length).toBeGreaterThan(0)
+
+      await browser.testHandle.expectBlob()
+
+      expect((await getSR()).events.length).toEqual(0)
+    })
+
+    it('Should stop recording if 429 response', async () => {
+      await browser.url(await browser.testHandle.assetURL('rrweb-instrumented.html', config()))
+        .then(() => browser.waitForAgentLoad())
+
+      expect(await getSR()).toEqual(expect.objectContaining({
+        events: expect.any(Array),
+        initialized: true,
+        recording: true,
+        mode: 1,
+        blocked: false
+      }))
+
+      await Promise.all([
+        browser.testHandle.expectBlob(),
```

Comment:
Is there a chance the initial SR harvest could happen before we get here?

Reply:
not on our test server.  Only way for that to happen is if the initial payload exceeds 64kb compressed and our pages are way too small to have that happen

Reply:
Should we try to make a test for pages that exceed the initial 64kb?

Reply:
good call.  we are only mocking this currently.  We should load a really big page for a better integration check

Reply:
added two tests, one for 64kb dom, and one for 1mb dom